### PR TITLE
Gifti: fixing load, and implementing write_gifti. And add tests mesh …

### PR DESCRIPTION
**Pull request by eolivi**
mostly for meshes coming from freesurfer

- fixing load_gifti (that was only for Column major, now both Row and Column).

- implementing write_gifti.

- Add tests mesh ……IOs.

### CODE REVIEW

**Comment by agramfort** [comment]
LGTM

File: OpenMEEG/libs/OpenMEEG/include/mesh.h [hunk 1]
```diff
@@ -225,19 +225,20 @@ namespace OpenMEEG {
...
         template <typename T>
         unsigned load_gifti(T, const bool&) {
             std::cerr << "You have to compile OpenMEEG with GIFTI to read GIFTI files" << std::endl;
             exit(1);
         }
-    #endif
-
         template <typename T>
         void save_gifti(T) const {
             std::cerr << "GIFTI writer : Not implemented" << std::endl;
```
Comment (papadop) [inline OpenMEEG/libs/OpenMEEG/include/mesh.h]: Same message as for load_gifti.

File: OpenMEEG/libs/OpenMEEG/src/mesh.cpp [hunk 2]
```diff
@@ -987,6 +1009,54 @@ namespace OpenMEEG {
         os.close();
     }
 
+    #ifdef USE_GIFTI
+    void Mesh::save_gifti(const std::string& filename) const {
+
+        int dims[2] = {(int)nb_vertices(), 3};
```
Comment (papadop) [inline OpenMEEG/libs/OpenMEEG/src/mesh.cpp]: static_cast

File: OpenMEEG/libs/OpenMEEG/src/mesh.cpp [hunk 3]
```diff
@@ -987,6 +1009,54 @@ namespace OpenMEEG {
...
+        gifti_image * gim = gifti_create_image(1, NIFTI_INTENT_POINTSET, NIFTI_TYPE_FLOAT32, 2, dims, 1);
+        gim->darray[0]->encoding = GIFTI_ENCODING_B64GZ;
+
+        // vertices
+        std::map<const Vertex *, int> map;
```
Comment (papadop) [inline OpenMEEG/libs/OpenMEEG/src/mesh.cpp]: I do not like the name map. Something less generic please.

File: OpenMEEG/libs/OpenMEEG/src/mesh.cpp [hunk 4]
```diff
@@ -987,6 +1009,54 @@ namespace OpenMEEG {
...
+        gim->darray[0]->data = new float[nb_vertices()*3];
+        float * pts = (float *)gim->darray[0]->data;
+        int i = 0;
+        for (const_vertex_iterator vit = vertex_begin(); vit != vertex_end(); ++vit, ++i) {
+            map[*vit] = i;
+            pts[3*i]   = float ((*vit)->x());
+            pts[3*i+1] = float ((*vit)->y());
+            pts[3*i+2] = float ((*vit)->z());
```
Comment (papadop) [inline OpenMEEG/libs/OpenMEEG/src/mesh.cpp]: constructors are not necessary here.

File: build_tools/travis/install.sh [hunk 5]
```diff
@@ -28,6 +29,7 @@ else
         -DENABLE_COVERAGE:BOOL=${USE_COVERAGE} \
         -DENABLE_PACKAGING:BOOL=${ENABLE_PACKAGING} \
         -DUSE_VTK:BOOL=${USE_VTK} \
+        -DUSE_GIFTI:BOOL=${USE_GIFTI} \
         -DUSE_CGAL:BOOL=${USE_CGAL} \
         -DBLASLAPACK_IMPLEMENTATION:BOOL=${BLASLAPACK_IMPLEMENTATION} \
         -DUSE_SYSTEM_clapack:BOOL=${USE_SYSTEM} \
```
Comment (papadop) [inline build_tools/travis/install.sh]: OK with the suppression of C casting and the removal of unnecessary {} on for loops.....

**Comment by agramfort** [comment]
@papadop merge if you're happy

**Pull request merged**